### ISSUE TITLE
Improve the error message in expect_offense

### ISSUE THREAD

**Issue by yayamochi** [rubocop/rubocop#12740]
Hi.
While creating a custom cop, I encountered a confusing error message.

## Is your feature request related to a problem? Please describe.
When using expect_offense and showing an error with `^^^^ {some_error}`, it is difficult to understand the error when the format is incorrect, so I would like to see an improvement.

cop/call_sample.rb
```ruby
module Cop
  class CallSample < RuboCop::Cop::Base
    MSG = 'sample error'
    RESTRICT_ON_SEND = [:sample]

    def on_send(node)
      add_offense(node)
    end
  end
end
```

spec/cop/call_sample_spec.rb
```ruby
  let(:cop) { Cop::CallSample.new }

  it 'good case' do
    # It's OK.
    expect_offense(<<~RUBY)
      some_object.sample
      ^^^^^^^^^^^^^^^^^^ sample error
    RUBY
  end

  it 'bad case' do
    # This raise error
    #  Failure/Error:
    #        expect_offense(<<~RUBY)
    #          some_object.sample
    #          ^^^^^^^^^^^^^^^^^^sample error
    #        RUBY
    #      
    #      RuntimeError:
    #        Use `expect_no_offenses` to assert that no offenses are found
    expect_offense(<<~RUBY)
      some_object.sample
      ^^^^^^^^^^^^^^^^^^sample error
    RUBY
  end
```

## Describe the solution you'd like
The 'bat case' error is not a no offenses error, but is caused by a formatting error in the ^^^^^ and missing spaces in the error message. So I would like you to show that.

## Describe alternatives you've considered
For example
```
RuntimeError:
    #        Leave a space between the ^ and the error message.
```

## Additional context
N/A
